frontend/src/lib/k8s/cluster.ts

```diff
@@ -36,6 +36,29 @@ export interface Cluster {
   [propName: string]: any;
 }
 
+export class BaseKubeObject {
+  static getBaseObject(): KubeObjectInterface {
+    return {
+      apiVersion: 'v1',
+      kind: '',
+      metadata: {
+        name: '',
+        namespace: '',
```

Comment:
This shouldn't be needed in the template because when the resource needs a namespace and doesn't have one, the default one for the cluster will be used.